### PR TITLE
feat(trace-view): Add the trace-view to event details

### PR DESCRIPTION
**Pull request by wmak**
- Shifts event id and created to the header so we can put quick trace on the top right
- Updated the pills to say `This {event type}` to hopefully make it a little less confusing when flipping between the types
- For now skipping trace-lite when its an error event, going to update the endpoint later to support errors
- Currently flagged so that the quick trace doesn't appear on error details without the summary flag
- Dropdown no longer appears when there's only 1 error
image

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/performance/quickTrace/utils.ts [hunk 1]
```diff
@@ -16,6 +17,21 @@ export function isTransaction(event: Event): event is EventTransaction {
...
 }
 
+/**
+ * An event can be an error or a transaction. We need to check whether the current
+ * event id is in the list of errors as well
+ */
+function isCurrentEvent(
+  event: TraceFull | QuickTraceEvent,
+  currentEvent: Event
+): boolean {
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/utils/performance/quickTrace/utils.ts:27]: Should this only check the errors array for the current event depending on the current event's type? Seems like avoidable computation.

File: src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx [hunk 2]
```diff
@@ -26,9 +27,8 @@ type Props = {
 };
 
 function EventMetas({event, organization, projectId, location, quickTrace}: Props) {
-  if (!isTransaction(event)) {
-    return null;
-  }
+  const type = isTransaction(event) ? 'transaction' : 'event';
+  const transaction = event as EventTransaction;
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx:31]: Is this explicit casting necessary? If necessary, you might be able to change `type === transaction` below to `isTransaction(event)` and Typescript might be able to pick up its an `EventTransaction`.

File: src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx [hunk 3]
```diff
@@ -83,15 +98,16 @@ function EventMetas({event, organization, projectId, location, quickTrace}: Prop
...
-  grid-template-columns: repeat(3, 1fr);
+  grid-template-columns: repeat(${p => (p.type === 'transaction' ? 3 : 2)}, 1fr);
   grid-template-rows: repeat(2, auto);
   grid-gap: ${space(2)};
   margin-bottom: ${space(2)};
 
   @media (min-width: ${p => p.theme.breakpoints[1]}) {
-    grid-template-columns: minmax(160px, 1fr) minmax(160px, 1fr) minmax(160px, 1fr) 6fr;
+    grid-template-columns: minmax(160px, 1fr) ${p =>
+        p.type === 'transaction' ? 'minmax(160px, 1fr)' : ''} minmax(160px, 1fr) 6fr;
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx:110]: I know this is going to be more LOC, but can we do something like
```javascript
${p => p.type === 'transaction' ? 'grid-template-columns: ...' : 'grid_template-columns: ...'}
```
this is probably a lot more legible.

File: src/sentry/static/sentry/app/views/performance/transactionDetails/quickTrace.tsx [hunk 4]
```diff
@@ -335,14 +331,19 @@ function EventNodeSelector({
...
+        });
+      }
     });
   });
+  // Filter out the current event so its not in the dropdown
+  events = currentEvent ? events.filter(e => e.event_id !== currentEvent.id) : events;
+
   let type: keyof Theme['tag'] = nodeKey === 'current' ? 'black' : 'white';
-  if (errors.length > 0) {
+  if (errors.length > 0 || (currentEvent && currentEvent.type !== 'transaction')) {
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/transactionDetails/quickTrace.tsx:346]:
```suggestion
  if (errors.length > 0 || (currentEvent?.type !== 'transaction')) {
```

File: src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx (hunk 2 above)
Comment (wmak) [reply src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx:31]: I had that originally but it kept beeping about the timestamps not existing, is there any other way to do `if event is EventTransaction`?

File: src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx [hunk 5]
```diff
@@ -48,29 +48,44 @@ function EventMetas({event, organization, projectId, location, quickTrace}: Prop
...
-        subtext={httpStatus}
-      />
+      {isTransaction(event) ? (
+        <MetaData
+          headingText={t('Total Duration')}
+          tooltipText={t(
+            'The total time elapsed between the start and end of this transaction.'
+          )}
+          bodyText={getDuration(
+            transaction.endTimestamp - transaction.startTimestamp,
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx:65]: You can use `event` here now that its been typed checked to be a `EventTransaction`.

File: src/sentry/static/sentry/app/utils/performance/quickTrace/utils.ts [hunk 6]
```diff
@@ -16,6 +17,22 @@ export function isTransaction(event: Event): event is EventTransaction {
...
+  return (
+    event.event_id === currentEvent.id ||
+    (!isTransaction(currentEvent) &&
+      event.errors !== undefined &&
+      event.errors.filter(e => e.event_id === currentEvent.id).length > 0)
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/utils/performance/quickTrace/utils.ts:32]: 🏌️
```javascript
!isTransaction(currentEvent) && event.errors?.some(e => e.event_id === currentEvent.id)
```

File: src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx (hunk 2 above)
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/transactionDetails/eventMetas.tsx:31]: You should be able to delete this now.

**Review by Zylphrex — APPROVED** [review]

File: src/sentry/static/sentry/app/views/performance/transactionDetails/quickTrace.tsx (hunk 4 above)
Comment (wmak) [reply src/sentry/static/sentry/app/views/performance/transactionDetails/quickTrace.tsx]: Realized I had to include this, cause currentEvent is only defined for the current Node, without the check meant that other nodes without errors would think they had an error.

File: src/sentry/static/sentry/app/utils/performance/quickTrace/utils.ts (hunk 6 above)
Comment (wmak) [reply src/sentry/static/sentry/app/utils/performance/quickTrace/utils.ts]: Didn't end up doing this cause of weird ts errors lol. But cleaned up the boolean by splitting into 2 ifs instead

**Pull request merged**